src/renderer/util/fileSystem.js

```diff
@@ -166,23 +140,36 @@ export const uploadImage = async (pathname, image, preferences) => {
       })
   }
 
-  const uploadByCliScript = async (filepath) => {
+  const uploadByCommand = async (uploader, filepath) => {
     let isPath = true
     if (typeof filepath !== 'string') {
       isPath = false
       const data = new Uint8Array(filepath)
       filepath = path.join(tmpdir(), +new Date())
       await fs.writeFile(filepath, data)
     }
-    cp.execFile(cliScript, [filepath], async (err, data) => {
-      if (!isPath) {
-        await fs.unlink(filepath)
-      }
-      if (err) {
-        return rj(err)
-      }
-      re(data.trim())
-    })
+    if (uploader === 'picgo') {
+      cp.exec(`picgo u "${filepath}"`, async (err, data) => {
+        if (!isPath) {
+          await fs.unlink(filepath)
+        }
+        if (err) {
+          return rj(err)
+        }
+
+        re(data.split('[PicGo SUCCESS]:')[1].trim())
```

Comment:
Can this be an error too or will the script always exit with non `0` on error?

Reply:
Ok, I'll extract `url` by RegExp.